### PR TITLE
add segment entity and releted objects

### PR DESCRIPTION
**Pull request by rprzedzik**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -106,7 +106,7 @@
             "Ergonode\\Value\\": "module/value/src",
             "Ergonode\\Workflow\\": "module/workflow/src",
             "Ergonode\\ConnectorMagento\\": "module/connector-magento/src",
-            "Ergonode\\TranslationDeepl\\": "module/translation-deepl/src"
+            "Ergonode\\TranslationDeepl\\": "module/translation-deepl/src",
```
Comment (BastekBielawski) [inline composer.json]: Remove ``,`` from end of line

File: module/fixture/src/Infrastructure/Faker/SegmentIdFaker.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * @param string|null $code
+     *
+     * @return SegmentId
+     *
+     * @throws \Exception
+     */
+    public function segmentId(?string $code = null): SegmentId
+    {
+
```
Comment (BastekBielawski) [inline module/fixture/src/Infrastructure/Faker/SegmentIdFaker.php]: Remove empty line

File: module/segment/src/Application/Controller/Api/SegmentController.php [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+     * )
+     * @SWG\Parameter(
+     *     name="segment",
+     *     in="path",
+     *     type="string",
+     *     description="Segment id",
+     * )
+     * @SWG\Response(
+     *     response=200,
+     *     description="Returns attribute",
```
Comment (wiewiurdp) [inline module/segment/src/Application/Controller/Api/SegmentController.php]: Wrong descirptions it should be "Returns segment"

File: module/segment/src/Application/Controller/Api/SegmentController.php [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+     * @SWG\Parameter(
+     *     name="body",
+     *     in="body",
+     *     description="Add segment",
+     *     required=true,
+     *     @SWG\Schema(ref="#/definitions/segment")
+     * )
+     * @SWG\Response(
+     *     response=200,
+     *     description="Returns attribute",
```
Comment (wiewiurdp) [inline module/segment/src/Application/Controller/Api/SegmentController.php]: Wrong descirptions it should be Return Id

File: module/segment/src/Domain/Entity/Segment.php [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    /**
+     * @var SegmentSpecificationInterface[]
+     */
+    private $specifications;
+
+    /**
+     * @param SegmentId          $id
+     * @param SegmentCode        $code
+     * @param TranslatableString $name
+     *
```
Comment (BastekBielawski) [inline module/segment/src/Domain/Entity/Segment.php]: Remove empty line

File: module/segment/src/Domain/ValueObject/SegmentCode.php [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    private $value;
+
+    /**
+     * @param string $value
+     */
+    public function __construct(string $value)
+    {
+        if (!self::isValid($value)) {
+            throw new \InvalidArgumentException('Invalid segment code value');
```
Comment (BastekBielawski) [inline module/segment/src/Domain/ValueObject/SegmentCode.php]: Add sprintf with value assigned

**Pull request merged**